src/anomalib/models/components/base/anomaly_module.py

```diff
@@ -39,7 +39,11 @@ class AnomalyModule(ExportMixin, pl.LightningModule, ABC):
     Acts as a base class for all the Anomaly Modules in the library.
     """
 
-    def __init__(self, post_processor: PostProcessor | None = None) -> None:
+    def __init__(
+        self,
+        post_processor: PostProcessor | None = None,
+        metrics: AnomalibMetric | Sequence[AnomalibMetric] | Evaluator | None = None,
```

Comment:
This was my initial design, but we decided to add the possibility of passing metrics directly for a bit more flexibility. 

> If the user uses AnomalibMetric | Sequence[AnomalibMetric] they already need to be aware of what an AnomalibMetric is. Since the init of Evaluator is quite simple, you wouldn't make it much harder and at the same time force them to be aware if they're setting val or test.

One drawback would be that the user needs to import and instantiate `Evaluator`, which would require a bit more knowledge of the code base than just being aware of `AnomalibMetric`. It also increases verbosity, as

```python
image_f1_score = F1Score(fields=["pred_label", "gt_label"])
image_auroc = AUROC(fields=["pred_label", "gt_label"])
Padim(metrics=[image_f1_score, image_auroc])
```

would become
```python
image_f1_score = F1Score(fields=["pred_label", "gt_label"])
image_auroc = AUROC(fields=["pred_label", "gt_label"])
evaluator = Evaluator(test_metrics=[image_f1_score, image_auroc])
Padim(evaluator=evaluator)
```

I do agree that it would simplify the signature of the constructor, and make the class definition more readable. It would also be more in line with the other submodules such as `PostProcessor`. 

@ashwinvaidya17 @samet-akcay any thoughts?

Reply:
Looking at the signature again... in terms of naming, which one would make more sense `Metrics` or `Evaluator`,

Reply:
strongly in favor of ```evaluator: Evaluator | None```

> One drawback would be that the user needs to import and instantiate `Evaluator`, which would require a bit more knowledge of the code base than just being aware of `AnomalibMetric`.

true, but i think it's fine as long as there is tutorial showing the minimal code and a few options (people will just cc cv from there)